File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        s = ""
        prefix = True
        
        while prefix and s != strs[0]:
            temp = s + strs[0][len(s)]
            for word in strs:
                if temp not in word[:len(temp)]:
                    prefix = False
                    break
            if prefix:
                s = temp
        
        return s

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    currPrefix = ""
    
    for (let i = 0; i < strs[0].length; i++){
        let temp = strs[0][i];
        let add = true;
        
        for (let j = 0; j < strs.length; j++){
            if (strs[j][i] !== temp){
                add = false;
                break;
            }
        }
        
        if (!add){
            break;
        } else{
            currPrefix += temp;
        }
    }
    
    return currPrefix;
};